cloudformation/cw-dashboard-substack.cfn.yaml

```diff
@@ -0,0 +1,554 @@
+
+              {%- for metrics_param in raid_metrics %}
+          - !Sub ',{"type":"metric","x":{{ coord.x }},"y":{{ coord.y }},"width":{{ graph_width }},"height":{{ graph_height }},"properties":{"view":"timeSeries","stacked":false,
+            {{ metrics_param.extra_params|join(',') }}{% if metrics_param.extra_params|length > 0 %},{% endif %}"region":"${AWS::Region}","metrics":'
+                {%- for i in range(number_of_raid_volumes) %}
+                  {%- set volume_loop = loop %}
+                  {%- for metric in metrics_param.metrics %}
+          - !Sub
+            - '{% if volume_loop.first and loop.first %}[{% else %},{% endif %}["AWS/EBS","{{ metric }}","VolumeId","${RAID_Volume{{ i|int +1 }}}"]{% if volume_loop.last and loop.last %}]}}{% endif %}'
+            - RAID_Volume{{ i|int +1 }}: !Select
+                - '{{ i|int }}'
+                - !Ref 'RAIDVolumesIds'
+                  {%- endfor %}
+                {%- endfor %}
+                {{- update_coord(graph_width, graph_height) -}}
+              {%- endfor %}
+
+              {#- Conditional RAID metrics #}
+              {%- set raid_metrics_conditions_params = [{'metric': 'VolumeConsumedReadWriteOps', 'supported_vol_types': ["io1"], 'extra_params': ['"title":"Consumed Read/Write Ops"']},
+                                                {'metric': 'VolumeThroughputPercentage', 'supported_vol_types': ["io1"], 'extra_params': ['"title":"Throughput Percentage"']},
+                                                {'metric': 'BurstBalance', 'supported_vol_types': ["gp2", "st1", "sc1"], 'extra_params': ['"title":"Burst Balance"']}]
+              %}
+              {%- for metric_condition_params in raid_metrics_conditions_params %}
+                {%- if raid_volume_type in metric_condition_params.supported_vol_types %}
+          - !Sub ',{"type":"metric","x":{{ coord.x }},"y":{{ coord.y }},"width":{{ graph_width }},"height":{{ graph_height }},"properties":{"view":"timeSeries","stacked":false,
+            {{ metric_condition_params.extra_params|join(',') }}{% if metric_condition_params.extra_params|length > 0 %},{% endif %}"region":"${AWS::Region}","metrics":'
+                  {%- for i in range(number_of_raid_volumes) %}
+          - !Sub
+            - '{{ '[' if loop.first else ',' }}["AWS/EBS","{{ metric_condition_params.metric }}","VolumeId","${RAID_Volume{{ i|int +1 }}}"]{{ ']}}' if loop.last else '' }}'
+            - RAID_Volume{{ i|int +1 }}: !Select
+                - '{{ i|int }}'
+                - !Ref 'RAIDVolumesIds'
+                  {%- endfor %}
+                  {{- update_coord(graph_width, graph_height) -}}
+                {%- endif %} {#- if raid_volume_type in metric_condition_params.supported_vol_types #}
+              {%- endfor -%}
+          {%- endif %} {#- if number_of_raid_volumes > 0 #}
+          {%- set efs_shared_dir = cfn_params.EFSOptions.split(',')[0] %}
```

Comment:
we should have the newline and the `{#- EFS metrics graphs #}` comment before this line, because it is the point on which we're starting evaluating EFS.

Reply:
Done